pvlib/albedo.py

```diff
@@ -0,0 +1,120 @@
+
+    Inputs to the model are the water color coefficient :math:`c` [-], the
+    water wave roughness coefficient :math:`r` [-] and the solar elevation
+    :math:`\alpha` [degrees]. Parameters are provided in [1]_ , and are coded
+    for convenience in :data:`~pvlib.albedo.WATER_COLOR_COEFFS` and
+    :data:`~pvlib.albedo.WATER_ROUGHNESS_COEFFS`. The values of these
+    coefficients are experimentally determined.
+
+    +------------------------+-------------------+-------------------------+
+    | Surface and condition  | Color coefficient | Wave roughness          |
+    |                        | (:math:`c`)       | coefficient (:math:`r`) |
+    +========================+===================+=========================+
+    | Clear water, no waves  | 0.13              | 0.29                    |
+    +------------------------+-------------------+-------------------------+
+    | Clear water, ripples   | 0.16              | 0.70                    |
+    | up to 2.5 cm           |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Clear water, ripples   | 0.23              | 1.25                    |
+    | larger than 2.5 cm     |                   |                         |
+    | (occasional whitecaps) |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Clear water,           | 0.30              | 2.00                    |
+    | frequent whitecaps     |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Green water, ripples   | 0.22              | 0.70                    |
+    | up to 2.5cm            |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Muddy water, no waves  | 0.19              | 0.29                    |
+    +------------------------+-------------------+-------------------------+
+
+    References
+    ----------
+    .. [1] Dvoracek M.J., Hannabas B. (1990). "Prediction of albedo for use in
+       evapotranspiration and irrigation scheduling." IN: Visions of the Future
+       American Society of Agricultural Engineers 04-90: 692-699.
+    """
+
+    if surface_condition is None:
+        if (color_coeff is None) or (wave_roughness_coeff is None):
+            raise ValueError('Either a `surface_condition` has to be chosen or'
+                             ' a combination of `color_coeff` and'
+                             '`wave_roughness_coeff`.')
+    else:
+        color_coeff = WATER_COLOR_COEFFS[surface_condition]
+        wave_roughness_coeff = WATER_ROUGHNESS_COEFFS[surface_condition]
+
+    solar_elevation = np.where(solar_elevation < 0, 0, solar_elevation)
+
+    albedo = color_coeff ** (wave_roughness_coeff * sind(solar_elevation) + 1)
+    return albedo
```

Comment:
It's possible to get an array out despite putting a Series in:

```python
In [399]: inland_water_dvoracek(pd.Series([45]), "muddy_water_no_waves")
Out[399]: array([0.13516185])
```

This is not an uncommon problem when using numpy tricks like `np.where`.  Our usual workaround is to fix up the types at the end of the function like this:

```python
    if isinstance(solar_elevation, pd.Series):
        albedo = pd.Series(albedo, index=solar_elevation.index)

    return albedo
```

A little inelegant, but it gets the job done.

Reply:
Wouldn't the type check of int/floats I suggested earlier do the job?

Reply:
Type checking has some tricky edge cases (e.g. `isinstance(np.array([45.])[0], (int, float))` is `False` because it is an `np.float64`, not a `float`).  A test based on `np.isscalar` might be ok.  

But I think it is generally better to fix up types at the end instead of following different calculation paths based on type.  Less chance of accidentally calculating different output values for different types.

Reply:
@kandersolar the problem is that `np.where` returns an array (even if you pass a pd.Series for the solar_elevation) - so the if statement is never true. I did a fix, but it might not be the prettiest one... can you approve it (or suggest an alternative)?